pkg/kubelet/cm/topologymanager/policy_test.go

```diff
@@ -1119,3 +1120,283 @@ func testPolicyMerge(policy Policy, tcases []policyMergeTestCase, t *testing.T)
 		}
 	}
 }
+
+func TestMaxOfMinAffinityCounts(t *testing.T) {
+	tcases := []struct {
+		hints    [][]TopologyHint
+		expected int
+	}{
+		{
+			[][]TopologyHint{},
+			0,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(), true},
+				},
+			},
+			0,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(0), true},
+				},
+			},
+			1,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(0, 1), true},
+				},
+			},
+			2,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(0, 1), true},
+					TopologyHint{NewTestBitMask(0, 1, 2), true},
+				},
+			},
+			2,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(0, 1), true},
+					TopologyHint{NewTestBitMask(0, 1, 2), true},
+				},
+				{
+					TopologyHint{NewTestBitMask(0, 1, 2), true},
+				},
+			},
+			3,
+		},
+		{
+			[][]TopologyHint{
+				{
+					TopologyHint{NewTestBitMask(0, 1), true},
+					TopologyHint{NewTestBitMask(0, 1, 2), true},
+				},
+				{
+					TopologyHint{NewTestBitMask(0, 1, 2), true},
+					TopologyHint{NewTestBitMask(0, 1, 2, 3), true},
+				},
+			},
+			3,
+		},
+	}
+
+	for _, tc := range tcases {
+		t.Run("", func(t *testing.T) {
+			result := maxOfMinAffinityCounts(tc.hints)
+			if result != tc.expected {
+				t.Errorf("Expected result to be %v, got %v", tc.expected, result)
+			}
+		})
+	}
+}
+
+func TestCompareHints(t *testing.T) {
+	tcases := []struct {
+		description                   string
+		bestNonPreferredAffinityCount int
+		current                       *TopologyHint
+		candidate                     *TopologyHint
+		expected                      string
+	}{
+		{
+			"candidate.NUMANodeAffinity.Count() == 0 (1)",
+			-1,
+			nil,
+			&TopologyHint{bitmask.NewEmptyBitMask(), false},
+			"current",
+		},
+		{
+			"candidate.NUMANodeAffinity.Count() == 0 (2)",
+			-1,
+			&TopologyHint{NewTestBitMask(), true},
+			&TopologyHint{NewTestBitMask(), false},
+			"current",
+		},
+		{
+			"current == nil (1)",
+			-1,
+			nil,
+			&TopologyHint{NewTestBitMask(0), true},
+			"candidate",
+		},
+		{
+			"current == nil (2)",
+			-1,
+			nil,
+			&TopologyHint{NewTestBitMask(0), false},
+			"candidate",
+		},
+		{
+			"!current.Preferred && candidate.Preferred",
+			-1,
+			&TopologyHint{NewTestBitMask(0), false},
+			&TopologyHint{NewTestBitMask(0), true},
+			"candidate",
+		},
+		{
+			"current.Preferred && !candidate.Preferred",
+			-1,
+			&TopologyHint{NewTestBitMask(0), true},
+			&TopologyHint{NewTestBitMask(0), false},
+			"current",
+		},
+		{
+			"current.Preferred && candidate.Preferred (1)",
```

Comment:
minor: I'd prefer a bit more descriptive name, for example mentioning here both hints are preferred.
Otherwise in few weeks people (surely myself) would be like "what was `current.Preferred && candidate.Preferred` case 1 again?"
 But I acknowledge the horizontal space for the output is limited, so up to you if you prefer to change or keep it this way